.github/workflows/testinparallel.yml

```diff
@@ -43,9 +42,24 @@ jobs:
         forge build --sizes
       id: build
 
-    - name: Run forge test for the file
-      run: forge test --match-path src/test/${{ matrix.file }} --no-match-contract FFI
+    - name: Run unit tests
+      run: forge test --no-match-contract Integration
+      env:
+        RPC_MAINNET: ${{ secrets.RPC_MAINNET }}
+        RPC_HOLESKY: ${{ secrets.RPC_HOLESKY }}
+        CHAIN_ID: ${{ secrets.CHAIN_ID }}
+
+    - name: Run integration tests
+      run: forge test --match-contract Integration
       env:
         RPC_MAINNET: ${{ secrets.RPC_MAINNET }}
         RPC_HOLESKY: ${{ secrets.RPC_HOLESKY }}
         CHAIN_ID: ${{ secrets.CHAIN_ID }}
+
+    - name: Run integration mainnet fork tests
+      run: forge test --match-contract Integration
```

Comment:
Perhaps we don't need to run the mainnet fork tests too frequently. Maybe on PR merges into `dev`?
```
      if: github.event_name == 'pull_request' && github.event.action == 'closed' && github.event.pull_request.merged && github.ref == 'refs/heads/dev'
```

Reply:
good flag if we run into more issues, but i think this should be fine for now